pubsub/rocketmq/rocketmq.go

```diff
@@ -16,120 +16,247 @@ package rocketmq
 	}
-	if r.metadata.ConsumerBatchSize != 0 {
-		opts = append(opts, mqc.WithPullBatchSize(int32(r.metadata.ConsumerBatchSize)))
+	if r.metadata.ConsumerGroupName != "" {
+		opts = append(opts, mqc.WithGroupName(r.metadata.ConsumerGroupName))
+	}
+	if r.metadata.NameServer != "" {
+		if strings.Contains(r.metadata.NameServer, ",") {
+			opts = append(opts, mqc.WithNameServer(strings.Split(r.metadata.NameServer, ",")))
+		} else if strings.Contains(r.metadata.NameServer, ";") {
+			opts = append(opts, mqc.WithNameServer(strings.Split(r.metadata.NameServer, ";")))
+		} else {
+			opts = append(opts, mqc.WithNameServer([]string{r.metadata.NameServer}))
+		}
 	}
 	if r.metadata.NameSpace != "" {
 		opts = append(opts, mqc.WithNamespace(r.metadata.NameSpace))
 	}
-	if r.metadata.Retries != 0 {
-		opts = append(opts, mqc.WithRetry(r.metadata.Retries))
-	}
 	if r.metadata.NameServerDomain != "" {
 		opts = append(opts, mqc.WithNameServerDomain(r.metadata.NameServerDomain))
 	}
-	if r.metadata.NameServer != "" {
-		opts = append(opts, mqc.WithNameServer(primitive.NamesrvAddr{r.metadata.NameServer}))
-	}
 	if r.metadata.AccessKey != "" && r.metadata.SecretKey != "" {
 		opts = append(opts, mqc.WithCredentials(primitive.Credentials{
-			AccessKey: r.metadata.AccessKey,
-			SecretKey: r.metadata.SecretKey,
+			AccessKey:     r.metadata.AccessKey,
+			SecretKey:     r.metadata.SecretKey,
+			SecurityToken: r.metadata.SecurityToken,
 		}))
 	}
-	return mq.NewPushConsumer(opts...)
+	if r.metadata.Retries > 0 {
+		opts = append(opts, mqc.WithRetry(r.metadata.Retries))
+	}
+	if r.metadata.ConsumerModel != "" {
+		switch strings.ToLower(r.metadata.ConsumerModel) {
+		case "broadcasting":
+			opts = append(opts, mqc.WithConsumerModel(mqc.BroadCasting))
+		case "clustering":
+			opts = append(opts, mqc.WithConsumerModel(mqc.Clustering))
+		default:
+			r.metadata.ConsumerModel = "Clustering"
+			opts = append(opts, mqc.WithConsumerModel(mqc.Clustering))
+			r.logger.Warnf("%s Consumer Model[%s] is error, expected [BroadCasting, Clustering], "+
+				"we will use default model [Clustering]", r.name, r.metadata.ConsumerModel)
+		}
+	}
+	if r.metadata.FromWhere != "" {
+		switch strings.ToLower(r.metadata.FromWhere) {
+		case "consumefromlastoffset":
+			opts = append(opts, mqc.WithConsumeFromWhere(mqc.ConsumeFromLastOffset))
+		case "consumefromfirstoffset":
+			opts = append(opts, mqc.WithConsumeFromWhere(mqc.ConsumeFromFirstOffset))
+		case "consumefromtimestamp":
+			opts = append(opts, mqc.WithConsumeFromWhere(mqc.ConsumeFromTimestamp))
+		default:
+			r.metadata.FromWhere = "ConsumeFromLastOffset"
+			opts = append(opts, mqc.WithConsumeFromWhere(mqc.ConsumeFromLastOffset))
+			r.logger.Warnf("%s Consumer FromWhere[%s] is error, "+
+				"expected [ConsumeFromLastOffset, ConsumeFromFirstOffset, ConsumeFromTimestamp], "+
+				"we will use default value [ConsumeFromLastOffset]", r.name, r.metadata.FromWhere)
```

Comment:
`"%s Consumer FromWhere[%s] is error, expected [ConsumeFromLastOffset, ConsumeFromFirstOffset, ConsumeFromTimestamp], we will use default value [ConsumeFromLastOffset]"

Reply:
ok